packages/automated-dispute/src/providers/protocolProvider.ts

```diff
@@ -238,34 +242,237 @@ export class ProtocolProvider implements IProtocolProvider {
-                    disputeId: "0x03" as DisputeId,
-                    dispute: {
-                        disputer: "0x12345678901234567890123456789012",
-                        proposer: "0x12345678901234567890123456789012",
-                        responseId: "0x02" as ResponseId,
-                        requestId: "0x01" as RequestId,
+    /**
+     * Decodes the log data for a specific event.
+     *
+     * @param eventName - The name of the event to decode.
+     * @param log - The log object containing the event data.
+     * @returns The decoded log data as an object.
+     * @throws {Error} If the event name is unsupported or if there's an error during decoding.
+     */
+    private decodeLogData<TEventName extends EboEventName>(
+        eventName: TEventName,
+        log: Log,
+    ): DecodedLogArgsMap[TEventName] {
+        let abi;
+        switch (eventName) {
+            case "RequestCreated":
+                abi = eboRequestCreatorAbi;
+                break;
+            case "ResponseProposed":
+                abi = oracleAbi;
+                break;
+            case "ResponseDisputed":
+                abi = oracleAbi;
+                break;
+            case "DisputeStatusChanged":
+                abi = oracleAbi;
+                break;
+            case "DisputeEscalated":
+                abi = oracleAbi;
+                break;
+            case "RequestFinalized":
+                abi = oracleAbi;
+                break;
+            default:
+                throw new UnsupportedEvent(`Unsupported event name: ${eventName}`);
+        }
+
+        try {
+            const decodedLog = decodeEventLog({
+                abi,
+                data: log.data,
+                topics: log.topics,
+                eventName,
+                strict: false,
+            });
+
+            return decodedLog.args as DecodedLogArgsMap[TEventName];
+        } catch (error) {
+            throw new DecodeLogDataFailure(error);
+        }
+    }
+
+    /**
+     * Parses an Oracle event log into an EboEvent.
+     *
+     * @param eventName - The name of the event.
+     * @param log - The event log to parse.
+     * @returns An EboEvent object.
+     */
+    private parseOracleEvent(eventName: EboEventName, log: Log) {
+        if (
+            ![
+                "ResponseProposed",
+                "ResponseDisputed",
+                "DisputeStatusChanged",
+                "DisputeEscalated",
+                "RequestFinalized",
+            ].includes(eventName)
+        ) {
+            throw new UnsupportedEvent(`Unsupported event name: ${eventName}`);
+        }
+
+        const baseEvent = {
+            name: eventName,
+            blockNumber: log.blockNumber ?? BigInt(0),
+            logIndex: log.logIndex ?? 0,
+            rawLog: log,
+            requestId: log.topics[1] as RequestId,
+        };
+
+        const decodedLog = this.decodeLogData(eventName, log);
```

Comment:
After setting `strict: true` while getting and decoding events, is it possible to do something like this to avoid using the switch statements?

```suggestion
        const decodedLog = this.decodeLogData(eventName, log);
        return {
            ...baseEvent,
            metadata: decodedLog
        }
```

I feel like there has to be a way to generalize this somehow, I might be wrong though.

Reply:
Could it be that the event's properties are prefixed with an underscore and we need to replicate those in our codebase?

Let's grab this event for example:
```
  event ResponseProposed(bytes32 indexed _requestId, bytes32 indexed _responseId, Response _response);
```

The corresponding type inside `DecodedLogArgsMap` is:
```
    ResponseProposed: {
        requestId: RequestId;
        responseId: string;
        response: string;
        blockNumber: bigint; // This property was removed yesterday
    };
```

Maybe refactoring those attributes by prefixing them with an underscore would comply with the ABI types? Feel free to do some quick validation on this.